firmware/application/apps/ui_settings.cpp

```diff
@@ -343,6 +343,31 @@ void SetUIView::focus() {
     button_save.focus();
 }
 
+/* SetSDCardView *********************************************/
+
+SetSDCardView::SetSDCardView(NavigationView& nav) {
+    add_children({&labels,
+                  &checkbox_sdcard_speed,
+                  &button_save,
+                  &button_cancel});
+
+    checkbox_sdcard_speed.set_value(pmem::config_sdcard_high_speed_io());
+
+    button_save.on_select = [&nav, this](Button&) {
+        pmem::set_config_sdcard_high_speed_io(checkbox_sdcard_speed.value());
```

Comment:
Personally I think this should call sdio_cclk_set(2) and do some file I/O _before_ saving the setting in Pmem.  If the firmware hangs when saving the setting, then the user should then know immediately that it doesn't work.  (It's rather inconvenient the steps required to turn off this setting if firmware hangs at power-up.)

Alternatively there could be a "Test" button, but I think testing automatically as part of the "Save" operation might be more fool-proof since the user might enable it without testing.

Reply:
I have various comportements on the device that is not working. It's not 'just not working'.
Sometimes the SD is read, but only parts.
I understand your point of view, but I think the few case when we would be able to detect fail from the code are rare, and I didn't had good results with the test implementations I made.
Anyway it will be open to modification,not only me :-)

Reply:
Ho I didn't see the test button proposal correctly.
Let me add it.

Reply:
Done.
There is a test button, which activate without saving to pmem.
You can then get back into the menu and try your luck with high speed on fileman (for example).
In case of  frozen/not really working, just restart the portapack and it should be OK.